File: WinForms/VB/Viewer/ExportForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms
Imports System.Data
Imports TatukGIS.NDK
Imports TatukGIS.NDK.WinForms

Namespace Viewer
    ''' <summary>
    ''' Summary description for WinForm1.
    ''' </summary>
    Public Class ExportForm
        Inherits System.Windows.Forms.Form
        ''' <summary>
        ''' Required designer variable.
        ''' </summary>
        Private components As System.ComponentModel.Container = Nothing
        Public grpSelectExtent As System.Windows.Forms.GroupBox
        Private WithEvents radioButton1 As System.Windows.Forms.RadioButton
        Private WithEvents radioButton2 As System.Windows.Forms.RadioButton
        Private WithEvents radioButton3 As System.Windows.Forms.RadioButton
        Private label1 As System.Windows.Forms.Label
        Public WithEvents cmbLayersList As System.Windows.Forms.ComboBox
        Private label2 As System.Windows.Forms.Label
        Public cmbShapeType As System.Windows.Forms.ComboBox
        Private btnCancel As System.Windows.Forms.Button
        Private btnOK As System.Windows.Forms.Button
        Public edtQuery As System.Windows.Forms.TextBox
        Private label3 As System.Windows.Forms.Label
        Private WithEvents btnCS As System.Windows.Forms.Button
        Private WithEvents label4 As System.Windows.Forms.Label
        Private WithEvents edCS As System.Windows.Forms.TextBox
        Public mainForm As WinForm
        Public CS As TGIS_CSCoordinateSystem

        Public Sub New()
            '
            ' Required for Windows Form Designer support
            '
            InitializeComponent()

            '
            ' TODO: Add any constructor code after InitializeComponent call
            '
        End Sub

        ''' <summary>
        ''' Clean up any resources being used.
        ''' </summary>
        Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
            If disposing Then
                If Not components Is Nothing Then
                    components.Dispose()
                End If
            End If
            MyBase.Dispose(disposing)
        End Sub

#Region "Windows Form Designer generated code"
        ''' <summary>
        ''' Required method for Designer support - do not modify
        ''' the contents of this method with the code editor.
        ''' </summary>
        Private Sub InitializeComponent()
            Me.grpSelectExtent = New System.Windows.Forms.GroupBox()
            Me.radioButton3 = New System.Windows.Forms.RadioButton()
            Me.radioButton2 = New System.Windows.Forms.RadioButton()
            Me.radioButton1 = New System.Windows.Forms.RadioButton()
            Me.label1 = New System.Windows.Forms.Label()
            Me.cmbLayersList = New System.Windows.Forms.ComboBox()
            Me.label2 = New System.Windows.Forms.Label()
            Me.cmbShapeType = New System.Windows.Forms.ComboBox()
            Me.btnCancel = New System.Windows.Forms.Button()
            Me.btnOK = New System.Windows.Forms.Button()
            Me.edtQuery = New System.Windows.Forms.TextBox()
            Me.label3 = New System.Windows.Forms.Label()
            Me.btnCS = New System.Windows.Forms.Button()
            Me.label4 = New System.Windows.Forms.Label()
            Me.edCS = New System.Windows.Forms.TextBox()
            Me.grpSelectExtent.SuspendLayout()
            Me.SuspendLayout()
            '
            'grpSelectExtent
            '
            Me.grpSelectExtent.Controls.Add(Me.radioButton3)
            Me.grpSelectExtent.Controls.Add(Me.radioButton2)
            Me.grpSelectExtent.Controls.Add(Me.radioButton1)
            Me.grpSelectExtent.Location = New System.Drawing.Point(11, 56)
            Me.grpSelectExtent.Name = "grpSelectExtent"
            Me.grpSelectExtent.Size = New System.Drawing.Size(230, 73)
            Me.grpSelectExtent.TabIndex = 0
            Me.grpSelectExtent.TabStop = False
            Me.grpSelectExtent.Text = "&Select extent"
            '
            'radioButton3
            '
            Me.radioButton3.Location = New System.Drawing.Point(8, 51)
            Me.radioButton3.Name = "radioButton3"
            Me.radioButton3.Size = New System.Drawing.Size(184, 17)
            Me.radioButton3.TabIndex = 2
            Me.radioButton3.Text = "Clippe&d by visible extent"
            '
            'radioButton2
            '
            Me.radioButton2.Location = New System.Drawing.Point(8, 33)
            Me.radioButton2.Name = "radioButton2"
            Me.radioButton2.Size = New System.Drawing.Size(184, 17)
            Me.radioButton2.TabIndex = 1
            Me.radioButton2.Text = "&Touched by visible extent"
            '
            'radioButton1
            '
            Me.radioButton1.Location = New System.Drawing.Point(8, 15)
            Me.radioButton1.Name = "radioButton1"
            Me.radioButton1.Size = New System.Drawing.Size(184, 17)
            Me.radioButton1.TabIndex = 0
            Me.radioButton1.Text = "&Whole extent"
            '
            'label1
            '
            Me.label1.Location = New System.Drawing.Point(11, 8)
            Me.label1.Name = "label1"
            Me.label1.Size = New System.Drawing.Size(181, 13)
            Me.label1.TabIndex = 1
            Me.label1.Text = "Select l&ayer to import from"
            '
            'cmbLayersList
            '
            Me.cmbLayersList.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
            Me.cmbLayersList.Location = New System.Drawing.Point(11, 24)
            Me.cmbLayersList.Name = "cmbLayersList"
            Me.cmbLayersList.Size = New System.Drawing.Size(230, 21)
            Me.cmbLayersList.TabIndex = 2
            '
            'label2
            '
            Me.label2.Location = New System.Drawing.Point(11, 136)
            Me.label2.Name = "label2"
            Me.label2.Size = New System.Drawing.Size(181, 13)
            Me.label2.TabIndex = 3
            Me.label2.Text = "Select s&hape type"
            '
            'cmbShapeType
            '
            Me.cmbShapeType.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList
            Me.cmbShapeType.Items.AddRange(New Object() {"Any supported shape", "Only Arcs (lines)", "Only Polygons (areas)", "Only Points (markers)", "Only Multipoints"})
            Me.cmbShapeType.Location = New System.Drawing.Point(11, 152)
            Me.cmbShapeType.Name = "cmbShapeType"
            Me.cmbShapeType.Size = New System.Drawing.Size(230, 21)
            Me.cmbShapeType.TabIndex = 4
            '
            'btnCancel
            '
            Me.btnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
            Me.btnCancel.Location = New System.Drawing.Point(128, 260)
            Me.btnCancel.Name = "btnCancel"
            Me.btnCancel.Size = New System.Drawing.Size(75, 25)
            Me.btnCancel.TabIndex = 5
            Me.btnCancel.Text = "&Cancel"
            '
            'btnOK
            '
            Me.btnOK.DialogResult = System.Windows.Forms.DialogResult.OK
            Me.btnOK.Location = New System.Drawing.Point(40, 260)
            Me.btnOK.Name = "btnOK"
            Me.btnOK.Size = New System.Drawing.Size(75, 25)
            Me.btnOK.TabIndex = 6
            Me.btnOK.Text = "&OK"
            '
            'edtQuery
            '
            Me.edtQuery.Location = New System.Drawing.Point(11, 192)
            Me.edtQuery.Name = "edtQuery"
            Me.edtQuery.Size = New System.Drawing.Size(230, 20)
            Me.edtQuery.TabIndex = 7
            '
            'label3
            '
            Me.label3.Location = New System.Drawing.Point(11, 176)
            Me.label3.Name = "label3"
            Me.label3.Size = New System.Drawing.Size(181, 13)
            Me.label3.TabIndex = 8
            Me.label3.Text = "&Query statement"
            '
            'btnCS
            '
            Me.btnCS.Location = New System.Drawing.Point(210, 234)
            Me.btnCS.Name = "btnCS"
            Me.btnCS.Size = New System.Drawing.Size(29, 20)
            Me.btnCS.TabIndex = 14
            Me.btnCS.Text = "..."
            Me.btnCS.UseVisualStyleBackColor = True
            '
            'label4
            '
            Me.label4.Location = New System.Drawing.Point(10, 218)
            Me.label4.Name = "label4"
            Me.label4.Size = New System.Drawing.Size(181, 13)
            Me.label4.TabIndex = 13
            Me.label4.Text = "&CS"
            '
            'edCS
            '
            Me.edCS.Location = New System.Drawing.Point(11, 234)
            Me.edCS.Name = "edCS"
            Me.edCS.Size = New System.Drawing.Size(195, 20)
            Me.edCS.TabIndex = 12
            '
            'ExportForm
            '
            Me.AcceptButton = Me.btnOK
            Me.AutoScaleDimensions = New System.Drawing.SizeF(96.0!, 96.0!)
            Me.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Dpi
            Me.ClientSize = New System.Drawing.Size(252, 290)
            Me.Controls.Add(Me.btnCS)
            Me.Controls.Add(Me.label4)
            Me.Controls.Add(Me.edCS)
            Me.Controls.Add(Me.label3)
            Me.Controls.Add(Me.edtQuery)
            Me.Controls.Add(Me.btnOK)
            Me.Controls.Add(Me.btnCancel)
            Me.Controls.Add(Me.cmbShapeType)
            Me.Controls.Add(Me.label2)
            Me.Controls.Add(Me.cmbLayersList)
            Me.Controls.Add(Me.label1)
            Me.Controls.Add(Me.grpSelectExtent)
            Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
            Me.Location = New System.Drawing.Point(288, 149)
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.Name = "ExportForm"
            Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
            Me.Text = "Export Layer"
            Me.grpSelectExtent.ResumeLayout(False)
            Me.ResumeLayout(False)
            Me.PerformLayout()

        End Sub
#End Region

        Private Sub ExportForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Dim i As Integer
            Dim ll As TGIS_Layer

            cmbLayersList.Items.Clear()
            ' add all layers of TGIS_LayerVector type to the list
            For i = mainForm.GIS.Items.Count - 1 To 0 Step -1
                ll = CType(mainForm.GIS.Items(i), TGIS_LayerAbstract)

                ' only vectors
                If TypeOf ll Is TGIS_LayerVector Then
                    cmbLayersList.Items.Add(ll.Name)
                End If
            Next i

            cmbLayersList.SelectedIndex = 0
            cmbShapeType.SelectedIndex = 0
            radioButton1.Checked = True
            If (Not CType(mainForm.GIS.Get(cmbLayersList.Text), TGIS_Layer) Is Nothing) Then
                CS = CType(mainForm.GIS.Get(cmbLayersList.Text), TGIS_Layer).CS
                edCS.Text = CS.WKT
            End If
        End Sub

        Private Sub radioButton1_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles radioButton1.CheckedChanged
            radioButton2.Checked = Not radioButton1.Checked
            radioButton3.Checked = Not radioButton1.Checked
        End Sub

        Private Sub radioButton2_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles radioButton2.CheckedChanged
            radioButton1.Checked = Not radioButton2.Checked
            radioButton3.Checked = Not radioButton2.Checked
        End Sub

        Private Sub radioButton3_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles radioButton3.CheckedChanged
            radioButton1.Checked = Not radioButton3.Checked
            radioButton2.Checked = Not radioButton3.Checked
        End Sub

        Public ReadOnly Property SelectedExtent() As Integer
            Get
                If radioButton1.Checked Then
                    Return 0
                ElseIf radioButton2.Checked Then
                    Return 1
                Else
                    Return 2
                End If
            End Get
        End Property

        Private Sub btnCS_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCS.Click
            Dim he As TGIS_HelpEvent = Nothing

            Dim dlg As TGIS_ControlCSSystem = New TGIS_ControlCSSystem()
            Try

                If (dlg.Execute(CS) = DialogResult.OK) Then

                    CS = dlg.CS
                    edCS.Text = CS.WKT
                End If

            Finally
                dlg = Nothing
            End Try

        End Sub

        Private Sub cmbLayersList_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmbLayersList.SelectedIndexChanged
            CS = CType(mainForm.GIS.Get(cmbLayersList.Text), TGIS_Layer).CS
            edCS.Text = CS.WKT
        End Sub
    End Class
End Namespace

File: WinForms/C#/Viewer/ExportForm.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using TatukGIS.NDK;
using TatukGIS.NDK.WinForms;

namespace Viewer
{
	/// <summary>
	/// Summary description for WinForm1.
	/// </summary>
	public class ExportForm : System.Windows.Forms.Form
	{
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;
		public  System.Windows.Forms.GroupBox grpSelectExtent;
		private System.Windows.Forms.RadioButton radioButton1;
		private System.Windows.Forms.RadioButton radioButton2;
		private System.Windows.Forms.RadioButton radioButton3;
		private System.Windows.Forms.Label label1;
		public  System.Windows.Forms.ComboBox cmbLayersList;
		private System.Windows.Forms.Label label2;
		public  System.Windows.Forms.ComboBox cmbShapeType;
		private System.Windows.Forms.Button btnCancel;
		private System.Windows.Forms.Button btnOK;
		public  System.Windows.Forms.TextBox edtQuery;
		private System.Windows.Forms.Label label3;
		public  WinForm mainForm;
        private TextBox edCS;
        private Label label4;
        private Button btnCS;
        public  TGIS_CSCoordinateSystem CS ;

		public ExportForm()
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose(bool disposing)
		{
			if (disposing)
			{
				if (components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose(disposing);
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.grpSelectExtent = new System.Windows.Forms.GroupBox();
            this.radioButton3 = new System.Windows.Forms.RadioButton();
            this.radioButton2 = new System.Windows.Forms.RadioButton();
            this.radioButton1 = new System.Windows.Forms.RadioButton();
            this.label1 = new System.Windows.Forms.Label();
            this.cmbLayersList = new System.Windows.Forms.ComboBox();
            this.label2 = new System.Windows.Forms.Label();
            this.cmbShapeType = new System.Windows.Forms.ComboBox();
            this.btnCancel = new System.Windows.Forms.Button();
            this.btnOK = new System.Windows.Forms.Button();
            this.edtQuery = new System.Windows.Forms.TextBox();
            this.label3 = new System.Windows.Forms.Label();
            this.edCS = new System.Windows.Forms.TextBox();
            this.label4 = new System.Windows.Forms.Label();
            this.btnCS = new System.Windows.Forms.Button();
            this.grpSelectExtent.SuspendLayout();
            this.SuspendLayout();
            // 
            // grpSelectExtent
            // 
            this.grpSelectExtent.Controls.Add(this.radioButton3);
            this.grpSelectExtent.Controls.Add(this.radioButton2);
            this.grpSelectExtent.Controls.Add(this.radioButton1);
            this.grpSelectExtent.Location = new System.Drawing.Point(11, 56);
            this.grpSelectExtent.Name = "grpSelectExtent";
            this.grpSelectExtent.Size = new System.Drawing.Size(230, 73);
            this.grpSelectExtent.TabIndex = 0;
            this.grpSelectExtent.TabStop = false;
            this.grpSelectExtent.Text = "&Select extent";
            // 
            // radioButton3
            // 
            this.radioButton3.Location = new System.Drawing.Point(8, 51);
            this.radioButton3.Name = "radioButton3";
            this.radioButton3.Size = new System.Drawing.Size(184, 17);
            this.radioButton3.TabIndex = 2;
            this.radioButton3.Text = "Clippe&d by visible extent";
            this.radioButton3.CheckedChanged += new System.EventHandler(this.radioButton3_CheckedChanged);
            // 
            // radioButton2
            // 
            this.radioButton2.Location = new System.Drawing.Point(8, 33);
            this.radioButton2.Name = "radioButton2";
            this.radioButton2.Size = new System.Drawing.Size(184, 17);
            this.radioButton2.TabIndex = 1;
            this.radioButton2.Text = "&Touched by visible extent";
            this.radioButton2.CheckedChanged += new System.EventHandler(this.radioButton2_CheckedChanged);
            // 
            // radioButton1
            // 
            this.radioButton1.Location = new System.Drawing.Point(8, 15);
            this.radioButton1.Name = "radioButton1";
            this.radioButton1.Size = new System.Drawing.Size(184, 17);
            this.radioButton1.TabIndex = 0;
            this.radioButton1.Text = "&Whole extent";
            this.radioButton1.CheckedChanged += new System.EventHandler(this.radioButton1_CheckedChanged);
            // 
            // label1
            // 
            this.label1.Location = new System.Drawing.Point(11, 8);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(181, 13);
            this.label1.TabIndex = 1;
            this.label1.Text = "Select l&ayer to import from";
            // 
            // cmbLayersList
            // 
            this.cmbLayersList.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.cmbLayersList.Location = new System.Drawing.Point(11, 24);
            this.cmbLayersList.Name = "cmbLayersList";
            this.cmbLayersList.Size = new System.Drawing.Size(230, 21);
            this.cmbLayersList.TabIndex = 2;
            this.cmbLayersList.SelectedIndexChanged += new System.EventHandler(this.cmbLayersList_SelectedIndexChanged);
            // 
            // label2
            // 
            this.label2.Location = new System.Drawing.Point(11, 136);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(181, 13);
            this.label2.TabIndex = 3;
            this.label2.Text = "Select s&hape type";
            // 
            // cmbShapeType
            // 
            this.cmbShapeType.DropDownStyle = System.Windows.Forms.ComboBoxStyle.DropDownList;
            this.cmbShapeType.Items.AddRange(new object[] {
            "Any supported shape",
            "Only Arcs (lines)",
            "Only Polygons (areas)",
            "Only Points (markers)",
            "Only Multipoints"});
            this.cmbShapeType.Location = new System.Drawing.Point(11, 152);
            this.cmbShapeType.Name = "cmbShapeType";
            this.cmbShapeType.Size = new System.Drawing.Size(230, 21);
            this.cmbShapeType.TabIndex = 4;
            // 
            // btnCancel
            // 
            this.btnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.btnCancel.Location = new System.Drawing.Point(133, 268);
            this.btnCancel.Name = "btnCancel";
            this.btnCancel.Size = new System.Drawing.Size(75, 25);
            this.btnCancel.TabIndex = 5;
            this.btnCancel.Text = "&Cancel";
            // 
            // btnOK
            // 
            this.btnOK.DialogResult = System.Windows.Forms.DialogResult.OK;
            this.btnOK.Location = new System.Drawing.Point(45, 268);
            this.btnOK.Name = "btnOK";
            this.btnOK.Size = new System.Drawing.Size(75, 25);
            this.btnOK.TabIndex = 6;
            this.btnOK.Text = "&OK";
            // 
            // edtQuery
            // 
            this.edtQuery.Location = new System.Drawing.Point(11, 192);
            this.edtQuery.Name = "edtQuery";
            this.edtQuery.Size = new System.Drawing.Size(230, 20);
            this.edtQuery.TabIndex = 7;
            // 
            // label3
            // 
            this.label3.Location = new System.Drawing.Point(11, 176);
            this.label3.Name = "label3";
            this.label3.Size = new System.Drawing.Size(181, 13);
            this.label3.TabIndex = 8;
            this.label3.Text = "&Query statement";
            // 
            // edCS
            // 
            this.edCS.Location = new System.Drawing.Point(13, 233);
            this.edCS.Name = "edCS";
            this.edCS.Size = new System.Drawing.Size(195, 20);
            this.edCS.TabIndex = 9;
            // 
            // label4
            // 
            this.label4.Location = new System.Drawing.Point(12, 217);
            this.label4.Name = "label4";
            this.label4.Size = new System.Drawing.Size(181, 13);
            this.label4.TabIndex = 10;
            this.label4.Text = "&CS";
            // 
            // btnCS
            // 
            this.btnCS.Location = new System.Drawing.Point(212, 233);
            this.btnCS.Name = "btnCS";
            this.btnCS.Size = new System.Drawing.Size(29, 20);
            this.btnCS.TabIndex = 11;
            this.btnCS.Text = "...";
            this.btnCS.UseVisualStyleBackColor = true;
            this.btnCS.Click += new System.EventHandler(this.btnCS_Click);
            // 
            // ExportForm
            // 
            this.AcceptButton = this.btnOK;
            this.AutoScaleDimensions = new System.Drawing.SizeF(96F, 96F);
            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Dpi;
            this.ClientSize = new System.Drawing.Size(252, 296);
            this.Controls.Add(this.btnCS);
            this.Controls.Add(this.label4);
            this.Controls.Add(this.edCS);
            this.Controls.Add(this.label3);
            this.Controls.Add(this.edtQuery);
            this.Controls.Add(this.btnOK);
            this.Controls.Add(this.btnCancel);
            this.Controls.Add(this.cmbShapeType);
            this.Controls.Add(this.label2);
            this.Controls.Add(this.cmbLayersList);
            this.Controls.Add(this.label1);
            this.Controls.Add(this.grpSelectExtent);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.Location = new System.Drawing.Point(288, 149);
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.Name = "ExportForm";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "Export Layer";
            this.Load += new System.EventHandler(this.ExportForm_Load);
            this.grpSelectExtent.ResumeLayout(false);
            this.ResumeLayout(false);
            this.PerformLayout();

		}
		#endregion

		private void ExportForm_Load(object sender, System.EventArgs e)
		{
			int i ;
			TGIS_Layer ll ;

			cmbLayersList.Items.Clear() ;
			// add all layers of TGIS_LayerVector type to the list
			for ( i = mainForm.GIS.Items.Count - 1; i >= 0; i-- )
			{
				ll = (TGIS_Layer)mainForm.GIS.Items[i] ;

				// only vectors
				if ( ll is TGIS_LayerVector )
					cmbLayersList.Items.Add( ll.Name ) ;
			}

			cmbLayersList.SelectedIndex = 0 ;
			cmbShapeType.SelectedIndex = 0 ;
			radioButton1.Checked = true ;
  
            if ((TGIS_Layer)mainForm.GIS.Get(cmbLayersList.Text) != null) {
                CS = ((TGIS_Layer)mainForm.GIS.Get(cmbLayersList.Text)).CS;
                edCS.Text = CS.WKT;
            };
		}

		private void radioButton1_CheckedChanged(object sender, System.EventArgs e)
		{
			radioButton2.Checked = !radioButton1.Checked ;
			radioButton3.Checked = !radioButton1.Checked ;
		}

		private void radioButton2_CheckedChanged(object sender, System.EventArgs e)
		{
			radioButton1.Checked = !radioButton2.Checked ;
			radioButton3.Checked = !radioButton2.Checked ;
		}

		private void radioButton3_CheckedChanged(object sender, System.EventArgs e)
		{
			radioButton1.Checked = !radioButton3.Checked ;
			radioButton2.Checked = !radioButton3.Checked ;
		}

		public int SelectedExtent
		{
			get
			{
				if (radioButton1.Checked)
					return 0 ;
				else if (radioButton2.Checked)
					return 1 ;
				else
					return 2 ;
			}
		}

        private void btnCS_Click(object sender, EventArgs e)
        {
            TGIS_ControlCSSystem dlg = new TGIS_ControlCSSystem();
            try
            {
                if (dlg.Execute(CS) == DialogResult.OK)
                {
                    CS = dlg.CS;
                    edCS.Text = CS.WKT;
                }
            }
            finally
            {
                dlg = null;
            }
        }

        private void cmbLayersList_SelectedIndexChanged(object sender, EventArgs e)
        {
            CS = ((TGIS_Layer)mainForm.GIS.Get(cmbLayersList.Text)).CS ;
            edCS.Text = CS.WKT;
        }
	}
}
